x-pack/plugins/alerting/server/saved_objects/is_rule_exportable.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger, SavedObject } from 'kibana/server';
+import { RawAlert } from '../types';
+import { AlertTypeRegistry } from '../alert_type_registry';
+
+export function isRuleExportable(
+  rule: SavedObject,
+  ruleTypeRegistry: AlertTypeRegistry,
+  logger: Logger
+): boolean {
+  const ruleSO = rule as SavedObject<RawAlert>;
+  try {
+    const ruleType = ruleTypeRegistry.get(ruleSO.attributes.alertTypeId);
+    if (!ruleType.isExportable) {
+      logger.warn(
+        `Skipping export of rule "${ruleSO.id}" because rule type "${ruleSO.attributes.alertTypeId}" is not exportable through this interface.`
+      );
```

Comment:
Do we feel one warning per rule is ok on export? or will this be too noisy when there are a lot of rules being exported?

Reply:
You are right, it might be noisy, especially if there are a lot of excluded rules (like the prebuilt security rules). Since `isExportable` is called on each saved object, there's no way of getting a count of the number excluded for logging purposes, although it will show up in the export.ndjson. I can remove the logging messages, WDYT?

Reply:
I found the message helpful, even if it did show up a few times because of all the Stack Monitoring alerts. It doesn't feel like noise as it only happens on an explicit user action so FWIW, I'm okay leaving it as is

Reply:
Ok, let's try it as is and adjust if we get any feedback 👍